docs/CNDP_README.md

```diff
@@ -0,0 +1,323 @@
+
+CNDP BESS port enables sending/receiving packets to/from network interface using AF-XDP.  CNDP integration with OMEC BESS UPF enables a software-based datapath and also provides deployment flexibility for kubernetes based deployments where CNDP uses [AF-XDP device plugin](https://github.com/intel/afxdp-plugins-for-kubernetes).
+
+Following are the steps required to build and test CNDP BESS UPF docker image:
+
+### Step 1: Build the OMEC UPF docker container
+
+> Note: If you are behind a proxy make sure to export/setenv http_proxy and https_proxy
+
+From the top level directory call:
+
+```
+$ make docker-build
+```
+
+### Step 2: Test setup
+
+Following diagram shows is a test setup. 
+
+![Test setup] (./images/cndp-omec-upf-test-setup.jpg)
+
+There are two systems: System 1 is runs CNDP BESS UPF and System 2 runs DPDK based packet generator which simulates traffic generated from multiple UEs and App servers. 
+
+Install NIC driver in System 1 from this link :  https://sourceforge.net/projects/e1000/files/ice%20stable/1.9.7/. This driver is used instead of in-tree kernel driver since this driver supports tc filter for creating queue groups and RSS for GTPU packet traffic steering. We need to compile and install this driver in your system with ADQ (Application Device Queues). As mentioned in the NIC driver README, use the following command to install the NIC driver.
+```
+$ sudo make -j CFLAGS_EXTRA='-DADQ_PERF_COUNTERS' install
+```
+
+Load the installed ice driver
+
+```
+sudo rmmod ice
+sudo modprobe ice
+```
+
+The setup uses Physical Function (PF) of PCIe network adaptor (where the driver supports AF-XDP zero copy ) for improved network I/O performance. AF-XDP zero copy support for SR-IOV driver and sub function support using devlink will be supported in future releases.
+
+System 1 and System 2 are connected using  physical links. Setup uses two network ports which represents access and core interface. This test setup uses Intel Ethernet 800 series network adapter (hereafter referred to as NIC).  NIC in System 1 has Intel Dynamics Device Personalization (DDP) for telecommunications workload enabled. DDP profile can help in GTPU packet traffic steering to required NIC hardware queues using RSS (Receive Side Scaling). DDP feature works along with XDP offload feature in NIC hardware to redirect GTPU packets directly to user space via AF-XDP sockets. Refer the Deployment section in this [document](https://builders.intel.com/docs/networkbuilders/intel-ethernet-controller-800-series-device-personalization-ddp-for-telecommunications-workloads-technology-guide.pdf) to enable DDP. Please follow Intel ethernet controller E810 DDP for telecommunications technology guide.
+
+### Step 3: Run UPF in System 1
+
+1) Setup hugepages in the system 1. For example, to setup 8GB of total huge pages (8 pages each of size 1GB in each NUMA node) using DPDK script, use the command `dpdk-hugepages.py -p 1G --setup 8G`
```

Comment:
This line assumes that user have already clone the DPDK repo. What about adding an easy way to "clone" only that script like (similar to what was done in [here](https://github.com/omec-project/upf/blob/master/docs/dpdk-configuration.md?plain=1#L29)):
```bash
wget https://raw.githubusercontent.com/DPDK/dpdk/main/usertools/dpdk-hugepages.py -O dpdk-hugepages.py
```